New mtl releases blogpost (#598)
3mdeb/news-and-ideas: blog/content/post/2024-07-23-new-mtl-releases.md

- [Artifacts in video playback in some players using HW acceleration](https://github.com/Dasharo/dasharo-issues/issues/948)
- [Only native resolution listed for internal panel](https://github.com/Dasharo/dasharo-issues/issues/949)

## Test coverage

The tests were performed on NovaCustom V540TU and V560TU platforms.
Checkout the results at our
[osfv-results repo](https://github.com/Dasharo/osfv-results):

- [V540TU](https://github.com/Dasharo/osfv-results/blob/main/boards/NovaCustom/MTL_14th_Gen/V540TU/v0.9.0-results.csv)
- [V560TU](https://github.com/Dasharo/osfv-results/blob/main/boards/NovaCustom/MTL_14th_Gen/V560TU/v0.9.0-results.csv)

The validation procedure is described at [laboratory assembly guide](https://docs.dasharo.com/transparent-validation/novacustom/laboratory-assembly-guide/#prerequisites).
Details about specific test cases can be found in our [Dasharo Test Specification]\
(<https://docs.dasharo.com/unified-test-documentation/overview/>)

## Our contributions

While working on the new release we have made multiple contributions
open source projects:

> ### Coreboot

Reviewer: ```suggestion
### coreboot
```

and all other instances if applicable

Author: Oh, the name is lowercase, I haven't noticed that, thanks! Fixed in [c50166d](https://github.com/3mdeb/news-and-ideas/pull/598/commits/c50166d3159eb2a2e5f7b67b13ce2c46a7329a46)

- <https://review.coreboot.org/c/coreboot/+/82671>
- <https://review.coreboot.org/c/coreboot/+/82672>
- <https://review.coreboot.org/c/coreboot/+/82673>
- <https://review.coreboot.org/c/coreboot/+/82686>
- <https://review.coreboot.org/c/coreboot/+/82685>
- <https://review.coreboot.org/c/coreboot/+/82674>
- <https://review.coreboot.org/c/coreboot/+/82898>

### Linux kernel

- <https://github.com/torvalds/linux/commit/e1c6db864599be341cd3bcc041540383215ce05e>

> 

Reviewer: We can add these 2 to the list:
- https://github.com/MrChromebox/edk2/commit/7f398d3b14928ffecbbe92bd93213db91dad7703
- https://github.com/systemd/systemd/commit/f5c8dd85ee13f9308498faf6a0c4837e604f8dcb

Author: added

The integrity of a firmware component is ensured thanks to cryptographic
signatures. See the link in section title for more details.

### [TPM Measured Boot](https://docs.dasharo.com/unified-test-documentation/dasharo-security/203-measured-boot/)

Measured Boot is another method of measuring the integrity of firmware components
by using Trusted Platform Module to store hashes of each firmware component in
PCR registers. Check the link in the header for more details.

### [Vboot recovery notification in UEFI Payload](https://docs.dasharo.com/unified-test-documentation/dasharo-security/201-verified-boot/)

Booting the device with a firmware component signed using wrong keys
will cause the device to boot into recovery mode. A notification
about that event will be displayed on the screen. For more details
check the link in the header.

![Vboot Verified Boot popup](/img/verified_boot_popup.png)

### [UEFI Shell](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/30P-uefi-shell/)

UEFI Shell is a command line interface that allows for interacting
with the UEFI firmware. It can be used for diagnosing, debugging,

> configuring and updating the firmware. Check the link in the header.

Reviewer: ```suggestion
and configuring the firmware. Check the link in the header.
```

Author: thanks!

![UEFI Shell](/img/uefi_shell_v2.2.png)

### [UEFI Secure Boot](https://docs.dasharo.com/unified-test-documentation/dasharo-security/206-secure-boot/)

Secure Boot is a fundamental security feature of UEFI specification.

> Thanks to secure boot, the device will only boot firmware components

Reviewer: ```suggestion
Thanks to secure boot, the device will only boot operating systems
```

which are trusted using cryptographic signatures embedded in the
firmware. Check the link in the header for more details.
![UEFI Secure Boot](/img/secure_boot.png)

### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

The feature allows updating the Embedded Controller firmware easily

> alongside updating the bios firmware.

Reviewer: ```suggestion
alongside updating the BIOS firmware.
```

### [BIOS boot medium write-protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

The recovery bios chip on your device is write-protected by default.
Modifying the firmware is only possible by entering the setup menu
and disabling the write protection explicitly or implicitly by using
Firmware Update Mode. Check the link in the header for more details
about Dasharo Security Options.

![BIOS boot medium write-protection](/img/bios_lock.png)

### [SMM BIOS write protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

With this option enabled, the BIOS can only be modified by
System Management Mode privileged code. It prevents flashing
the firmware with programs like [flashrom](https://www.flashrom.org/).
![SMM BIOS write protection](/img/smm_bios_write_protection.png)

### [Early boot DMA protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

The IOMMU DMA protection enabled early
in the POST process prevents Direct Memory Access attacks.
Leaking secrets and injecting malware using devices such

> as PCIe cards and USB4/thunderbolt devices is prevented

Reviewer: ```suggestion
as PCIe cards and USB4/Thunderbolt devices is prevented
```

thanks to this feature.

### [Early Sign of Life display output](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/347-sign-of-life/)

The Early Sign of Life display is a feature that displays the
firmware version during the boot phase. It is useful for
diagnosing boot issues and verifying that the firmware
is up-to-date.

### [Current limiting for USB-PD power supplies](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31H-usb-type-c/#utc020001-usb-type-c-pd-current-limiting-ubuntu-2204)

Limiting the current draw from USB-Power Delivery is an
important safety feature that controls the maximum current

> that can flow through the USB-PD. This prevents damaging the device

Reviewer: ```suggestion
that can flow through the USB-PD power supply. This prevents damaging the device
```

as well as what it is connected to in case of any malfunction

> like short-circuiting the USB port.

Reviewer: Not really, this is just about respecting the limits that the USB-C charger tells us. Before, the laptop would just pull as many watts as it wanted from a charger, overdrawing from it. Most chargers would engage overcurrent protection at this point and reset or shutdown, but some cheaper chargers could be damaged if this was done repeatedly.

Author: [5d83859](https://github.com/3mdeb/news-and-ideas/pull/598/commits/5d838599cca210aea0fc9571724be1a9c17bb9b1)

### [Setup menu password configuration](https://docs.dasharo.com/dasharo-menu-docs/overview/$user-password-management)

Setting up a password for the setup menu allows for
protecting the configuration from unauthorized changes.

### [Wi-Fi / Bluetooth module disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

With this option enabled, the Wi-Fi and Bluetooth modules
are powered off and disabled. Use this when you want to
ensure that the device is not transmitting any data wirelessly.
![Wi-Fi / Bluetooth module disable option](/img/enable_wifi_bt.png)

### [Built-in webcam disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

With this option disabled the power to the integrated webcam
is cut off making it as good as physically removed from the device
until the camera is enabled again.
![Webcam disable option](/img/enable_camera.png)

### [USB stack disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#usb-configuration)

> The option controls loading of USB firmware USB drivers. Disabling

Reviewer: ```suggestion
The option controls loading of firmware USB drivers. Disabling
```

---
title: New Dasharo v0.9.0 Meteor Lake releases
abstract: "Dasharo v0.9.0 for 14th gen Intel Meteor Lake has just been released
            bringing numerous new features and improvements. Check out
            what\'s new!"
cover: /covers/novacustom-dasharo-v0.9.0.png
author: filip.golas
layout: post
published: true
date: 2024-07-23
archives: "2024"

tags:
  - dasharo
  - coreboot
  - intel
  - meteorlake
  - linux
  - contribution
categories:
  - Firmware
  - Security

---

## Introduction

We are thrilled to announce that the support for newest Intel
Meteor Lake platform has been added to Dasharo in release v0.9.0!
The newest release features numerous new features and improvements
related to security, functionality and quality of life.
Thanks to everyone who contributed to this release!

## Added features

Here is a brief introduction to the changes in Dasharo v0.9.0,

> for more details check out:

Author: [e278f32](https://github.com/3mdeb/news-and-ideas/pull/598/commits/e278f325514e50e83528873c1d8c1bbef13a7cb0)

### [Vboot recovery notification in UEFI Payload](https://docs.dasharo.com/unified-test-documentation/dasharo-security/201-verified-boot/)

Booting the device with a firmware component signed using wrong keys
will cause the device to boot into recovery mode. A notification
about that event will be displayed on the screen. For more details
check the link in the header.

![Vboot Verified Boot popup](/img/verified_boot_popup.png)

### [UEFI Shell](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/30P-uefi-shell/)

UEFI Shell is a command line interface that allows for interacting
with the UEFI firmware. It can be used for diagnosing, debugging,
and configuring the firmware. Check the link in the header.
![UEFI Shell](/img/uefi_shell_v2.2.png)

### [UEFI Secure Boot](https://docs.dasharo.com/unified-test-documentation/dasharo-security/206-secure-boot/)

Secure Boot is a fundamental security feature of UEFI specification.
Thanks to secure boot, the device will only boot operating systems
which are trusted using cryptographic signatures embedded in the
firmware. Check the link in the header for more details.
![UEFI Secure Boot](/img/secure_boot.png)

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Author: I was just copying the new features from https://docs.dasharo.com/variants/novacustom_v540tu/releases/#v090-2024-07-18

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Reviewer 2: please removed this one @philipandag .
@mkopec  please check if there are any features that need to be removed and let's publish this blog post.

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Reviewer: I don't see any more removed features

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Author: Should I also update the release notes on docs.dasharo.com?

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Author: removed: [252e077](https://github.com/3mdeb/news-and-ideas/pull/598/commits/252e07736c4ad1e934185b25d0e21ed0aadc4617)

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Author: Removed: [252e077](https://github.com/3mdeb/news-and-ideas/pull/598/commits/252e07736c4ad1e934185b25d0e21ed0aadc4617)

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Reviewer: > Should I also update the release notes on docs.dasharo.com?

yes please :)

> ### [Automatic Embedded Controller update](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31G-ec-and-superio/#ecr031001-ec-firmware-sync-in-coreboot)

Author: PR in docs https://github.com/Dasharo/docs/pull/877

The feature allows updating the Embedded Controller firmware easily
alongside updating the BIOS firmware.
Check out the link in the section title for more details.

> ### [Firmware update mode](https://docs.dasharo.com/guides/firmware-update/#firmware-update-mode)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

The Firmware Update Mode is a quick way of ensuring your device
is prepared for a firmware update. It temporarily changes the
configuration and returns it to the previous state after
the update is completed. Check the link in the header for more details
and instructions on how to use it.
![Firmware Update Mode Confirmation Screen](/img/setup_menu_fum_confirmation.png)

> ### [BIOS boot medium write-protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

Reviewer: That's the way the option is called, we should try to stay consistent. @philipandag please fix

> ### [BIOS boot medium write-protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

The recovery bios chip on your device is write-protected by default.
Modifying the firmware is only possible by entering the setup menu
and disabling the write protection explicitly or implicitly by using
Firmware Update Mode. Check the link in the header for more details
about Dasharo Security Options.

![BIOS boot medium write-protection](/img/bios_lock.png)

### [SMM BIOS write protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

With this option enabled, the BIOS can only be modified by
System Management Mode privileged code. It prevents flashing
the firmware with programs like [flashrom](https://www.flashrom.org/).
![SMM BIOS write protection](/img/smm_bios_write_protection.png)

> ### [Early boot DMA protection](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

The Early Sign of Life display is a feature that displays the
firmware version during the boot phase. It is useful for
diagnosing boot issues and verifying that the firmware
is up-to-date.

### [Current limiting for USB-PD power supplies](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/31H-usb-type-c/#utc020001-usb-type-c-pd-current-limiting-ubuntu-2204)

Limiting the current draw from USB-Power Delivery is an
important safety feature that controls the maximum current
that can flow through the USB-PD power supply.
With this feature the device will limit how much current it
draws not to exceed the power supplies specifications.
Most chargers would engage overcurrent protection if the
device tried to draw too much current, but triggering
it repeatedly could shorten the lifespan of the charger
or simply damage it.

### [Setup menu password configuration](https://docs.dasharo.com/dasharo-menu-docs/overview/$user-password-management)

Setting up a password for the setup menu allows for
protecting the configuration from unauthorized changes.

> ### [Wi-Fi / Bluetooth module disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

With this option enabled, the Wi-Fi and Bluetooth modules
are powered off and disabled. Use this when you want to
ensure that the device is not transmitting any data wirelessly.
![Wi-Fi / Bluetooth module disable option](/img/enable_wifi_bt.png)

> ### [Built-in webcam disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#dasharo-security-options)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

With this option disabled the power to the integrated webcam
is cut off making it as good as physically removed from the device
until the camera is enabled again.
![Webcam disable option](/img/enable_camera.png)

> ### [USB stack disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#usb-configuration)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

The option controls loading of firmware USB drivers. Disabling
it will prevent the USB devices from working until an OS
with a proper USB driver is loaded.
![Enable USB stack](/img/enable_usb_stack.png)

> ### [Network stack disable option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#networking-options)

Reviewer: fixed in d90da1784289f180135eb1bb38dd84fbaf5f36ff

of the battery and possibly slow down it's degradation
with time.
Check the link in the header for more details.
![Battery threshold options](/img/battery_threshold.png)

### [Intel ME disable option in setup menu](https://docs.dasharo.com/osf-trivia-list/me/)

This option allows you to disable the Intel Management Engine
which is a subsystem that has full access to the system, including
system memory and network. It is active at any time and
is transparent to the OS. ME is proprietary software and so
it is a potential security risk.
If you don't need it, you can disable it using this option.
![Intel ME disable option](/img/intel_me_enable.png)

### [Block boot when battery is too low](https://docs.dasharo.com/unified-test-documentation/dasharo-compatibility/359-boot-blocking/#test-cases-common-documentation)

With this feature enabled, the device will not boot if the battery
is below a certain threshold. The reason for this it to prevent
a situation where the charge would become too low
to support the boot process and the device would unexpectedly shut down
which could lead to data loss or corruption.
![Battery block popup](/img/battery_block_popup.jpg)

> ### [Power on AC option in setup menu](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/#power-management-options)

Author: To my defence, the titles here are also the same as in https://docs.dasharo.com/variants/novacustom_v540tu/releases/#v090-2024-07-18

![Throttling temperature adjustment](/img/cpu_throttling_threshold.png)

## Known issues

Sadly there are some issues that were not resolved in time for the release.
Here is a list of the issues on which we are working on:

- [No HDMI output in FW on V540TU and V560TU](https://github.com/Dasharo/dasharo-issues/issues/930)
- [Laggy behaviour on Manjaro (KDE) with open drivers](https://github.com/Dasharo/dasharo-issues/issues/911)
- [V540TU: Option Previous power state restoration doesn't work](https://github.com/Dasharo/dasharo-issues/issues/931)
- [Artifacts in video playback in some players using HW acceleration](https://github.com/Dasharo/dasharo-issues/issues/948)
- [Only native resolution listed for internal panel](https://github.com/Dasharo/dasharo-issues/issues/949)

## Test coverage

The tests were performed on NovaCustom V540TU and V560TU platforms.
Checkout the results at our
[osfv-results repo](https://github.com/Dasharo/osfv-results):

- [V540TU](https://github.com/Dasharo/osfv-results/blob/main/boards/NovaCustom/MTL_14th_Gen/V540TU/v0.9.0-results.csv)
- [V560TU](https://github.com/Dasharo/osfv-results/blob/main/boards/NovaCustom/MTL_14th_Gen/V560TU/v0.9.0-results.csv)

> The validation procedure is described at [laboratory assembly guide](https://docs.dasharo.com/transparent-validation/novacustom/laboratory-assembly-guide/#prerequisites).

Reviewer 2: 404, as it should be. We decided not to use this setup anymore.

Author: It is, again, copied directly from the releases page. https://docs.dasharo.com/variants/novacustom_v540tu/releases/#v090-2024-07-18 Do I delete this part?

Reviewer 2: yes, please

Author: @BeataZdunczyk [1ad7b30](https://github.com/3mdeb/news-and-ideas/pull/598/commits/1ad7b30522e1139e769986e08325f1794dcaf88a)
